crates/oxc_linter/src/rules/import/no_amd.rs

```diff
@@ -0,0 +1,111 @@
+use oxc_ast::ast::{Expression, Argument};
+use oxc_diagnostics::{
+  miette::{self, Diagnostic},
+  thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{Atom, ModuleKind, Span};
+use oxc_ast::AstKind;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint-plugin-import(no-amd): Do not use AMD `require` and `define` calls.")]
+#[diagnostic(severity(warning), help("Expected imports instead of AMD {1}()"))]
+struct NoAmdDiagnostic(#[label] pub Span, Atom);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoAmd;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Forbid AMD `require` and `define` calls.
+    ///
+    /// ### Example
+    ///
+    /// ```javascript
+    /// // fail
+    /// require([a, b], function() {} );
+    /// // pass
+    /// require('../name');
+    /// require(`../name`);
+    /// ```
+    NoAmd,
+    nursery
+);
+
+/// https://github.com/import-js/eslint-plugin-import/blob/main/src/rules/no-amd.js
+impl Rule for NoAmd {
+  fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+    let _ = match ctx.source_type().module_kind() {
```

Comment:
It seems I have a misunderstanding of `ctx.source_type().module_kind()`. The original code was `if (context.getScope().type !== 'module') { return; }`. Does OXC has any function similar to the original one? Current code is incorrect and will cause test case `function x() { define(["a"], function (a) {}) }` fail, which should pass.

Reply:
Try compare the `ctx.scopes().root_scope_id()` and `node.scope_id()`. if they are the same, this means the CallExpression is in the top level.